modules/EnsEMBL/Web/Component/Gene/ComparaOrthologs.pm

```diff
@@ -0,0 +1,138 @@
+=head1 LICENSE
+
+Copyright [2009-2024] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package EnsEMBL::Web::Component::Gene::ComparaOrthologs;
+
+use strict;
+use warnings;
+
+sub create_gene_tree_links {
+  my $self = shift;
+  my $params = shift;
+
+  my $hub          = $self->hub;
+  my $species_defs = $hub->species_defs;
+  my $species = $params->{species};
+  my $stable_id = $params->{stable_id};
+  my $orthologue = $params->{orthologue};
+  my $cdb = $params->{cdb};
+
+  my $is_pan = $cdb =~/compara_pan_ensembl/;
+  my $current_species_clusterset_ids = $self->get_current_species_clusterset_ids($species_defs);
+  my $comparison_species_clusterset_ids = $self->get_clusterset_ids_for_comparison_species($species_defs, $species);
+
+  my $common_clusterset_ids = $self->get_common_clusterset_ids(
+    $current_species_clusterset_ids,
+    $comparison_species_clusterset_ids
+  );
```

Comment:
Given the clusterset_ids of the query gene, this allows us to filter them by the clusterset_ids of the orthologue:
```suggestion
  my $common_clusterset_ids = [grep { $orthologue->{'homologue'}->has_GeneTree($_) } @$current_gene_clusterset_ids];
```